src/Compilers/Core/Portable/InternalUtilities/HashAlgorithms.cs

```diff
@@ -12,9 +12,12 @@ namespace Roslyn.Utilities
 
     internal abstract class HashAlgorithm : IDisposable
     {
-        private static readonly MethodInfo s_bytesMethod;
-        private static readonly MethodInfo s_bytesOffsetCountMethod;
-        private static readonly MethodInfo s_streamMethod;
+        private static readonly MethodInfo s_ComputeHash_bytes_Method;
+        private static readonly MethodInfo s_ComputeHash_bytesOffsetCount_Method;
+        private static readonly MethodInfo s_ComputeHash_stream_Method;
+        private static readonly MethodInfo s_TransformBlock_Method;
+        private static readonly MethodInfo s_TransformFinalBlock_Method;
+        private static readonly MethodInfo s_Hash_PropertyGetter;
 
```

Comment:
I wish. I'm told that invoking a method via reflection is 1000x slower than a regular method invocation.

On the Java platform a method invoked via reflection is only 2-3x slower (after the first few invocations).